common/djangoapps/entitlements/models.py

```diff
@@ -24,3 +54,65 @@ class CourseEntitlement(TimeStampedModel):
         help_text='The current Course enrollment for this entitlement. If NULL the Learner has not enrolled.'
     )
     order_number = models.CharField(max_length=128, null=True)
+    _policy = models.ForeignKey(CourseEntitlementPolicy, null=True, blank=True)
+
+    @property
+    def expired_at_datetime(self):
+        if not self.expired_at and self.get_days_until_expiration() < 0:
+            self.expired_at = datetime.utcnow()
+            self.save()
+        return self.expired_at
+
+    @expired_at_datetime.setter
+    def expired_at_datetime(self, value):
+        self.expired_at = value
+
+    @property
```

Comment:
Would it make sense to use `@cached_property` so that we don't need to instantiate multiple `CourseEntitlementPolicy` objects?